internal/auth/auth.go

```diff
@@ -22,66 +24,113 @@ var privToAPI = map[privileges.Privilege]model.Privilege{
 	}
+
 	return next(ctx)
 }
 
-type PrivilegeContractValidator struct {
-	VehicleNFTAddress common.Address
-	// ManufAddr         string
+// ManufacturerNFTPrivCheck checks if the claim set in the context includes the correct address the required privileges for the ManufacturerNFT contract.
+func (pcv *PrivilegeContractValidator) ManufacturerNFTPrivCheck(ctx context.Context, obj interface{}, next graphql.Resolver, priv model.Privilege) (res interface{}, err error) {
+	if err := pcv.checkPrivWithAddress(ctx, []model.Privilege{priv}, pcv.manufacturerNFTAddress); err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnauthorized.Error(), err)
+	}
+
+	return next(ctx)
 }
 
-// VehicleNFTPrivCheck checks if the claim set in the context includes the correct address the required privileges for the VehicleNFT contract.
-func (pcv *PrivilegeContractValidator) VehicleNFTPrivCheck(ctx context.Context, _ any, next graphql.Resolver, privs []model.Privilege) (any, error) {
+func (pcv *PrivilegeContractValidator) checkPrivWithAddress(ctx context.Context, privs []model.Privilege, contractAddr common.Address) error {
 	claim, err := getTelemetryClaim(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("%s: %w", errUnauthorized.Error(), err)
+		return err
 	}
 
-	if claim.ContractAddress != pcv.VehicleNFTAddress {
-		return nil, fmt.Errorf("%w: contract address mismatch", errUnauthorized)
+	if claim.ContractAddress != contractAddr {
+		return fmt.Errorf("contract address mismatch")
 	}
 
 	for _, priv := range privs {
 		if _, ok := claim.privileges[priv]; !ok {
-			return nil, fmt.Errorf("%w: missing required privilege %s", errUnauthorized, priv)
+			return fmt.Errorf("missing required privilege %s", priv)
 		}
 	}
 
+	return nil
+}
+
+// VehicleTokenCheck checks if the vehicle tokenID in the context matches the tokenID in the claim.
+func (pcv *PrivilegeContractValidator) VehicleTokenCheck(ctx context.Context, _ any, next graphql.Resolver) (any, error) {
+	claim, err := getTelemetryClaim(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	vehicleTokenID, err := getRequestValues[int](ctx, "tokenId")
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnauthorized.Error(), err)
+	}
+
+	if strconv.Itoa(vehicleTokenID) != claim.TokenID {
+		return nil, fmt.Errorf("%w: tokenID mismatch", errUnauthorized)
+	}
+
 	return next(ctx)
 }
 
-// func (pcv *PrivilegeContractValidator) ManufacturerNFTPrivCheck(ctx context.Context, _ any, next graphql.Resolver, privs []model.Privilege) (any, error) {
-// 	claim, err := getTelemetryClaim(ctx)
-// 	if err != nil {
-// 		return nil, fmt.Errorf("%s: %w", errUnauthorized.Error(), err)
-// 	}
+func (pcv *PrivilegeContractValidator) ManufacturerTokenCheck(ctx context.Context, obj interface{}, next graphql.Resolver) (res interface{}, err error) {
+	claim, err := getTelemetryClaim(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO allow user to search by addr, tokenID, or serial
```

Comment:
We should implement this TODO since we allow for tokenId and serial queries.

Reply:
I converted back to a draft.

Honestly I would just do address for now.